js/build.js

```diff
@@ -33,20 +33,41 @@ Fliplet.Widget.instance('sso-saml', function(data) {
       throw new Error('Provider ' + ssoProviderPackageName + ' has not registered on Fliplet.Widget.register with an "authorize()" function.');
     }
 
-    ssoProvider
-      .authorize(data)
-      .then(function onAuthorized() {
-        $btn.addClass('hidden');
-        $('.sso-confirmation').fadeIn(250, function() {
-          setTimeout(function() {
-            Fliplet.Navigate.to(data.redirectAction);
-          }, 100);
+    $btn.text('Please wait...').addClass('disabled');
+
+    ssoProvider.authorize(data).then(function onAuthorized() {
+      return Fliplet.UI.Toast({
+        position: 'bottom',
+        backdrop: true,
+        tapToDismiss: false,
+        duration: false,
+        message: 'Verifying your login...'
+      }).then(function (toast) {
+        return Fliplet.Session.passport('saml2').data().then(function (response) {
+          var user = response.user;
```

Comment:
@squallstar Are the data such as email, firstName and lastName etc. subject to change? If so, then perhaps the ID (I assume it's a unique identifier) should be sufficient?

Reply:
Though, if the ID is unique, the SAML session is also only unique to the endpoint where people have logged into, right?

Otherwise, endpoint A and endpoint B could both potentially have the same user ID.

Reply:
@tonytlwu none of the keys returned here are guaranteed to be here. We could have one, or all four. It depends on what they expose from their side. Following your point I'd recommend keeping at least "id" and "email". For your records, this is where the mapping (data normalisation) is done https://github.com/Fliplet/fliplet-api/blob/master/libs/passports/saml2.js#L116-L121

Reply:
@squallstar Perhaps a simple logic of We will store this list of keys... is insufficient to meet th objective. For me the objective is The object should store as little as necessary to uniquely identify the user. I'm therefore thinking the following:

If the response has an ID, store the ID
Else, if the response has an email, store the email
Else, (I'm not sure what...)
It should store the SAML login source because different sources could share the same ID

Reply:
@squallstar In some ways, yes. I can't imagine a **Like** social feature being used to collect likes from multiple organisations. However, these will help us run analysis such as, "How many app users in our system are engaging with the social features" and so on.

What sort of information do we have that we can use to identify the specific SAML provider/endpoint?

Reply:
And do you have any concerns of adding such information?

Reply:
@tonytlwu unfortunately nothing, because apps can only have one instance of SAML2 and this is managed by an app component (behind iframed as a provider in the SAML2 Login component interface).

Therefore, the unique field is most likely going to be `appId`. Do you agree?

Reply:
Let's use `type: "saml2"`, `id`/`email`, `region` and `organizationId` then. I think that’ll help us identify the unique SAML2 user more efficiently.